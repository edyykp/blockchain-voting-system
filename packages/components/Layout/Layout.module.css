.logoWrapper {
  height: 1em;
  margin-left: 0.5rem;
}

/* export const theme = {
  colors: {
    primary: '',
    secondary: '',
    greys: {
      white: '#FFFFFF',
      lighter: '#F2F2F2',
      light: '#CCCCCC',
      grey: '#8C8C8C',
      dark: '#666666',
      darker: '#333333',
    },
  },
}; */
