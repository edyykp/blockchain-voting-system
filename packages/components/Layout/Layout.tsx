import React, { useState } from 'react';
import Head from 'next/head';

import { Navbar, Sidebar } from '@packages/core';
import { useSiteProperties } from '@packages/config';

import { Footer } from '../Footer';

type LayoutProps = {
  children: React.ReactNode;
};

export const Layout = ({ children }: LayoutProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const valueOf = useSiteProperties();

  const toggle = () => {
    setIsOpen(!isOpen);
  };

  const text = {
    title: valueOf('site_title'),
    description: valueOf('site_description'),
  };
  return (
    <>
      <Head>
        <title>{text.title}</title>
        <meta name="description" content={text.description} />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Navbar toggle={toggle} />
      <Sidebar isOpen={isOpen} toggle={toggle} />
      {children}
      <Footer />
    </>
  );
};
