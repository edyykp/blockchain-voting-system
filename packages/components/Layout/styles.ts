import styled, { createGlobalStyle } from 'styled-components';

export const LogoWrapper = styled.div`
  height: 1em;
  margin-left: 0.5rem;
`;

export const GlobalStyle = createGlobalStyle`
html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

`;

export const theme = {
  colors: {
    primary: '',
    secondary: '',
    greys: {
      white: '#FFFFFF',
      lighter: '#F2F2F2',
      light: '#CCCCCC',
      grey: '#8C8C8C',
      dark: '#666666',
      darker: '#333333',
    },
  },
};
