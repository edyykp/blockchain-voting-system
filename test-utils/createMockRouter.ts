import { NextRouter } from 'next/router';

export function createMockRouter(router: Partial<NextRouter>): NextRouter {
  return {
    basePath: '',
    pathname: '/',
    route: '/',
    asPath: '/',
    query: {},
    isFallback: false,
    back: jest.fn(),
    beforePopState: jest.fn(),
    prefetch: jest.fn(),
    push: jest.fn(),
    reload: jest.fn(),
    replace: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isLocaleDomain: false,
    isReady: false,
    defaultLocale: 'en',
    domainLocales: [],
    isPreview: false,
    ...router,
  };
}
