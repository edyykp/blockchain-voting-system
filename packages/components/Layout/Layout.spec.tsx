import { render } from '@testing-library/react';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import '@testing-library/jest-dom';

import { Layout } from './Layout';
import { createMockRouter } from 'test-utils/createMockRouter';

const setup = () => {
  return {
    ...render(
      <RouterContext.Provider value={createMockRouter({})}>
        <Layout>
          <div>This is a test</div>
        </Layout>
      </RouterContext.Provider>,
    ),
  };
};

describe('Layoutâ†’', () => {
  it('should render Navbar', () => {});

  it('should render children', () => {
    const { container, getByText } = setup();

    expect(getByText('This is a test')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render navigation bar', () => {
    const { getByTestId } = setup();

    expect(getByTestId('navigation-bar')).toBeInTheDocument();
  });
});
