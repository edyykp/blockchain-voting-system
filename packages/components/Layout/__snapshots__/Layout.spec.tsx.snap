// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout→ should render children 1`] = `
<div>
  <nav
    class=""
    data-testid="navigation-bar"
  >
    <div
      class="container"
      data-testid="navigation-bar-logo"
    >
      <a
        class="logo "
        href="/"
      >
        BVS
      </a>
      <div
        class="mobileMenuIcon"
        data-testid="mobile-menu-icon"
      >
        <svg
          class="faBars"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </div>
      <ul
        class="menu"
        data-testid="navigation-bar-menu"
      >
        <li
          class="menuItem"
        >
          <a
            class="menuUrls "
          >
            About
          </a>
        </li>
        <li
          class="menuItem"
        >
          <a
            class="menuUrls "
          >
            Data Protection
          </a>
        </li>
      </ul>
      <div
        class="ctaContainer"
        data-testid="navigation-bar-cta"
      >
        <a
          class="button  primary
      false
      mediumButton
      false
      "
        >
          Sign in
        </a>
      </div>
    </div>
  </nav>
  <aside
    class="container "
    data-testid="sidebar"
  >
    <div
      class="iconContainer"
    >
      <svg
        class="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </div>
    <div
      class="menuWrapper"
    >
      <ul
        class="menu"
      >
        <a
          class="link"
        >
          About
        </a>
        <a
          class="link"
        >
          Data Protection
        </a>
      </ul>
    </div>
    <div
      class="ctaWrapper"
    >
      <a
        class="button  primary
      false
      mediumButton
      false
      "
      >
        Sign in
      </a>
    </div>
  </aside>
  <div>
    This is a test
  </div>
  <footer
    class="footer"
    data-testid="footer"
  >
    <ul
      class="menu"
    >
      <a
        class="link"
      >
        About
      </a>
      <a
        class="link"
      >
        Data Protection
      </a>
    </ul>
    <div
      class="copyright"
    >
      © 2022, Proudly Coded by 
      Eduard-Constantin Stoica
    </div>
  </footer>
</div>
`;
